### PR TITLE
Shazbot's Launchers and sprites

### PR DESCRIPTION
**Pull request by shazbot194**
17hs4wwfq6gprqdishxlnw
Hello, I am shazbot, also know in servers as P.A.T.C.H. I here have some new sprites and two new guns for your consideration. The resprites also have some minor function with them, but I can disable these at your word. None of these sprites are final, but should be an improvement over the current ones.

The two new guns are the grenade launcher and rocket launcher. They both shoot very deadly ammo. The grenade launcher shoots special 40mm grenades that show what is loaded into the chamber. These 40mm grenades are not compatible with the old 40mm grenades. There are currently three grenades, flashbang, teargas, and hight explosive (HE). Flashbang is marked white and has the same effect as a flashbang. Teargas is red and has the same effect as a teargas grenade. HE is blue and is a normal 40mm explosive round.
The rocket launcher acts very similarly, it can load four different rockets, and unload them as well. The standard rocket has no stripe, and is a weak explosive, doing as much damage as a 40mm HE grenade. The blue striped rocket that looks like the standard rocket is a HE rocket, and has a much bigger blast radius and thus does way more damage. The pointed rocket with a red stripe is a HEAT grenade. It has a weak explosive, but does massive impact damage, meant to be used to take down mechs. The black grenade with a blue stripe is a EMP rocket, also meant to take down mechs.
The three weapons I have resprited are the tommy gun, the enforcer, and the WT-550. The enforcer has recived a full treatment of alt skins, suppressors, and flashlights. The Wt-550 can now be suppressed in addition to looking good. The tommy gun just looks overall better in my opinion.

I hope I didn't forget anything, and like I said, I am more then willing to change anything you want.

As a side note, these do not have in-hand sprites just yet, depending on how long this takes to merge, I can either make them into this PR, or make a new PR.

### CODE REVIEW

**Comment by MrFroztee** [comment]
Just make a note to eventually implement some in_hand sprites, should work for us for now, once the conflicts are handled.

**Comment by msgerbs** [comment]
How are these even conflicting? These files haven't been changed in forever. Is your fork up to date with SheepFighter/Stratus?

File: code/modules/projectiles/projectile/reusable.dm [hunk 1]
```diff
@@ -66,3 +66,23 @@
...
+	ammo_type = /obj/item/weapon/grenade/flashbang
+	range = 10
+	var/det_time = 0
+
+/obj/item/projectile/bullet/reusable/flash/handle_drop()
+	if(!dropped)
+		var/obj/item/weapon/grenade/F = new ammo_type(loc)
+		dropped = TRUE
+		spawn(det_time)
+			F.prime()
```
Comment (msgerbs) [inline code/modules/projectiles/projectile/reusable.dm:85]: The rocket just spawns a flashbang grenade? There must be a better way to do this.

**Review by msgerbs — COMMENTED** [review]
I'm torn on how the flashbag/teargas rockets are written. On one hand it's a pretty slick way to make it easy to add new rockets for any pre-existing grenade, but it feels kind of hackish. Thoughts, @SheepFighter?

**Comment by shazbot194** [comment]
Closed due to a working PR having been made

**Comment by msgerbs** [comment]
[Grenade launchers need work](https://trello.com/c/yJgwamCG/10-grenade-launchers-need-work)

**Pull request closed without merging**